nf_core/modules/pipeline_modules.py

```diff
@@ -208,39 +107,107 @@ def install(self, module=None):
             return False
-        log.debug("Installing module '{}' at modules hash {}".format(module, self.modules_repo.modules_current_hash))
-
         # Set the install folder based on the repository name
         install_folder = ["nf-core", "software"]
         if not self.modules_repo.name == "nf-core/modules":
             install_folder = ["external"]
 
+        # Load 'modules.json'
+        modules_json_path = os.path.join(self.pipeline_dir, "modules.json")
+        with open(modules_json_path, "r") as fh:
+            modules_json = json.load(fh)
+
+        current_version = modules_json["modules"].get(module)
+
+        if current_version is not None:
+            # Fetch the latest commit for the module
+            git_log = get_module_git_log(module, per_page=1, page_nbr=1)
+            if len(git_log) == 0:
+                log.error(f"Was unable to fetch version of module '{module}'")
+                return False
+            latest_version = git_log[0]["git_sha"]
+            if current_version == latest_version:
+                log.info("Already up to date")
+                return True
+            elif not force:
+                log.error("Found newer version of module. To install use '--force'")
+                return False
+
+            # Check for flag '--latest'
+            if latest:
+                version = latest_version
+            else:
+                try:
+                    version = prompt_module_version_sha(module, installed_sha=current_version["git_sha"])
+                except SystemError as e:
+                    log.error(e)
+        else:
+            # Check for flag '--latest'
+            if latest:
+                # Fetch the latest commit for the module
+                git_log = get_module_git_log(module, per_page=1, page_nbr=1)
+                if len(git_log) == 0:
+                    log.error(f"Was unable to fetch version of module '{module}'")
+                    return False
+                version = git_log[0]["git_sha"]
+            else:
+                try:
+                    version = prompt_module_version_sha(module)
+                except SystemError as e:
+                    log.error(e)
+                    sys.exit(1)
+
         # Check that we don't already have a folder for this module
         module_dir = os.path.join(self.pipeline_dir, "modules", *install_folder, module)
         if os.path.exists(module_dir):
-            log.error("Module directory already exists: {}".format(module_dir))
-            # TODO: uncomment next line once update is implemented
-            # log.info("To update an existing module, use the commands 'nf-core update'")
-            return False
+            if not force:
+                log.error(
+                    "Module directory already exists but module {} is not present in 'module.json'".format(module_dir)
+                )
+                return False
```

Comment:
Here it would be nice to ask the user whether to overwrite (we should have something similar somewhere else, I think `nf-core modules create-test-yml`).
Basically a simple question whether to force overwrite and if yes, set 'force' to true.